### ISSUE TITLE
Within-local market analysis

### ISSUE THREAD

**Issue by huntallcott** [huntallcott/FoodDeserts#6]
The goal of this issue is to prepare the analysis of healthy eating by income conditional on store, in the new data structure. @handbury I think will take the lead.

Jessie, I have created a file called $Externals/Calculations/Homescan/HHxYearxStore.dta, which has key household-year-store and key variable being Health Index. Note that $Externals is a file path that you should define for your computer in GitHub/FoodDeserts/Code/FoodDesertsMasterFile.do

For now I suggest using HealthIndex_per1000Cal as the main healthy eating measure. We can put other health measures into all datasets using Code/DataPrep/Homescan/InsertHealthMeasures.do.

I think that the main analysis here could just be graphical, perhaps something that matches the figures produced by Code/Analysis/StylizedFacts/StylizedFacts.do? 

Let me know if questions or thoughts?

**Comment by handbury** [comment]
Thanks Hunt.  

I agree that keeping the analysis here graphical and as in-line with what your draft already has so as to minimize re-drafting work.  I plan on:
1. adding a set of points to the current figure 2 but using data residualized from hh-zip fixed effects
2. replicating stylized fact figure 2 for the HH-year-store level data and adding to these figures a set of points using data residualized from store fixed effects 

I'll let you know if I have any specific questions as I get into coding in your file structure/using git.

**Comment by handbury** [comment]
@huntallcott: is there a document that outlines which datasets are created in what code - or an easy way for me to figure this out?  For example, it looks like Figure 2 is created in GitHub\FoodDeserts\Code\Analysis\StylizedFacts\StylizedFacts.do using $Externals/Calculations/Homescan/NominalIncome.dta.  Which do file creates $Externals/Calculations/Homescan/NominalIncome.dta?

Thanks,

Jessie

**Comment by huntallcott** [comment]
@handbury:
Thanks for jumping into this. There is a semi-easy way to figure this out, using two pieces of info.
1. As described [here](https://github.com/huntallcott/lab/wiki/Folder-Organization), the folders are generally (although due to my errors, sometimes imperfectly) aligned, so that dta files in Calculations/Homescan/ are created by do files in Code/DataPrep/Homescan/.
2. The entire routine runs from the file FoodDesertsMasterFile.do. Starting from that do file, then you can go to Code/DataPrep/DataPrepMasterFile.do and follow how Homescan is prepared.

If you track that through you'll see that the NominalIncome.dta file is created by Code/DataPrep/Homescan/TransactionDataPrep.do. 

In keeping with that system: for these figures, I suggest leaving ADD figure 2 as it is in StylizedFacts.do. I would suggest you create a new do file called something like Code/Analysis/ReducedForm/WithinLocationAnalysis.do that creates the new figures, and outputs the figures as pdfs to GitHub/FoodDeserts/Output/ReducedForm/.  Then please add to Code/Analysis/ReducedForm/ReducedFormSection.do this line: include Code/Analysis/ReducedForm/WithinLocationAnalysis.do. Then when we run FoodDesertsMasterFile.do for replication, your figures will be created.

Sorry if this is confusing, and I appreciate your willingness to use this system. Let me know if other questions?

**Comment by handbury** [comment]
@huntallcott Thanks for the background on the coding structure (you might have directed us there before, sorry if I missed it).

In any case, I did as you suggested and started a WithinLocationAnalysis program.  The plots (as well as the data underlying them, if we want to reference std errors, for example) go to Analysis/Output/ReducedForm/Figures.  

A few points worth noting:

- I define a market in time and space.  So the residualized data are residualized from zip/store x year FEs. Note that the intermediate results, also residualized from year FEs don't overlap perfectly (as a result of the widening of the gap documented in ADD) but the charts are a bit messier with those included.

- I normalize the means from their value lowest income group.  I find it easier to see the reduction in the gap that results from residualizing from market fixed effects.  So the pattern of the raw data will be the same as in Figure 2 but the y-axis values will be shifted.

- The store level results are noisier, not surprisingly.  In case this has to do with the fact that we're observing purchases that aren't reflective of households' entire purchase baskets, I replicated the analysis using the purchases in only the store that each household spends 50% or more of their total expenditure (in any given year).  These plots are much less noisy (see the plots with the _minexpsh50 suffixes).

- I haven't residualized these means from household size, since ADD FIgure 2 doesn't but it might be worth doing so.  I don't think that we should residualize from any demographics that are more obvious SES predictors, though. 

-  In HRS, we found that the original nutrition-SES gradient is quite a bit smaller for the "identifying" sample in the within-tract analysis (that is, the sample of households, in the HRS case, that resided in tracts that contain both college and non-college households).  The "identifying" sample was much more representative for the within-store analysis, though.  I expect this will also be the case here, so if the focus of this analysis in the merged paper is the "within-store" stuff, then it's probably fine to leave it for robustness/a footnote.  In any case, we'd want to think about what restriction to place on the "identifying" sample (e.g., a store-year where we observe households in more than one income group shopping?)

- The HHxYearxStore data was missing the whole grain (Whole) and InSample variables.  So the InSample restrictor isn't used for the within-store analysis and it isn't yet run for whole grains.

Let me know if you have questions/comments.  

[MacroNutrients_Income_zip_code_x_yr.pdf](https://github.com/huntallcott/FoodDeserts/files/2085367/MacroNutrients_Income_zip_code_x_yr.pdf)
[Healthfulness_Income_store_code_uc_x_yr.pdf](https://github.com/huntallcott/FoodDeserts/files/2085368/Healthfulness_Income_store_code_uc_x_yr.pdf)
[Healthfulness_Income_store_code_uc_x_yr_minexpsh50.pdf](https://github.com/huntallcott/FoodDeserts/files/2085369/Healthfulness_Income_store_code_uc_x_yr_minexpsh50.pdf)
[Healthfulness_Income_zip_code_x_yr.pdf](https://github.com/huntallcott/FoodDeserts/files/2085370/Healthfulness_Income_zip_code_x_yr.pdf)
[MacroNutrients_Income_store_code_uc_x_yr.pdf](https://github.com/huntallcott/FoodDeserts/files/2085371/MacroNutrients_Income_store_code_uc_x_yr.pdf)
[MacroNutrients_Income_store_code_uc_x_yr_minexpsh50.pdf](https://github.com/huntallcott/FoodDeserts/files/2085372/MacroNutrients_Income_store_code_uc_x_yr_minexpsh50.pdf)

**Comment by huntallcott** [comment]
Stunning figures. Amazing how closely these line up. 
I changed the output directory, I hope that's OK.

Do you want to write in the text of this into the beginning of Section 4 of the lyx file (before the other two "reduced form" analyses) so that we can see how it looks together? 
NB the lyx file won't compile until I change the figure file paths to the GitHub system, I'll try to do that ASAP.

**Comment by handbury** [comment]
@huntallcott Yes, I'll write this up by early next week.  Let me know if you have any thoughts on the issues I flagged above (or can add the whole grain variable to the household-year-store data). 

@mollyschnell Hope that the travel home was smooth.  Just wanted to flag these new results (see the thread above). Hunt's updated the data, so we've implemented the cross-sectional analysis (up to, but not including the "identifying sample" restriction) and it all looks pretty consistent with what we had.  

I might take at presenting our time series in the same vein as the event study next week.  Presumably that will go on another issue thread though.

**Comment by handbury** [comment]
Preliminary write-up is now in Output/Lyx/WithinLocationAnalysis.  I'm not sure where the rest of the draft is so please move around as necessary.  

@huntallcott and @mollyschnell - One issue that is worth us discussing:  the results here look different to the HRS results.  The "raw" reduction in the gap when controlling for zip fixed effects looks similar, but the results are different when we limit ourselves to the "identifying" sample.   I implemented this "identifying sample" restriction here by limiting our attention to zip-years or store-years for which we saw households in more than one income group.  Formally, this is the identification sample (the income bin fixed effects are the same in this sample as in the full sample).  This isn't very restrictive though.  It only reduces the sample size by 12% for the zip-FE analysis - relative to around 50% in the tract-FE analysis in HRS where we required that both college and non-college households lived in the tract. As a result, the raw means hardly change between the full and identifying samples and the reduction in the gap from the "identifying sample" baseline remains larger when including zip fixed effects (around 25%) than when including store fixed effects (around 10%).  In HRS we found the zip FEs resulted in a smaller reduction in the gap than the store FEs, which we thought made sense since HHs have the ability to sort more across stores than residential zip codes. Perhaps this intuition was incorrect? 

We have a couple of options here:
1. Look at whether our intuition on sorting across stores vs. zip codes was wrong in HRS.
2. Drop the within-zip analysis.  This is my preference. a) The 10% number we get from the within-store analysis is is more in line with what we are observing elsewhere (both the structural analysis and the HRS time series). b) The within-store number is presented as the "preferred" result in HRS in any case, since households in the same tract have different access to cars and can still live in different locations. These differences in access are likely larger in zip codes which can vary dramatically in size and population. c) The benefit of the within-zip analysis was that we could look at the healthfulness of the overall household bundle. Perhaps that is less important here given the addition of the structural model.

@huntallcott and @mollyschnell Let me know your thoughts and I'll re-write accordingly.

**Comment by handbury** [comment]
@huntallcott FYI: Homescan panelist tracts are saved in Dropbox under Externals\Data\Nielsen\Homescan\CensusTracts\from_usda_2002_2011

**Comment by handbury** [comment]
Let me see if we have beyond 2010 elsewhere.  I think that we inferred 2011 household tracts from their 2010 ones though.

On Jun 12, 2018, at 6:24 PM, Hunt Allcott <notifications@github.com<mailto:notifications@github.com>> wrote:


@handbury<https://github.com/handbury> I see only through 2010, is that correct?
Also I deleted the 2002 and 2003 files; I presume that's OK with you.

**Comment by huntallcott** [comment]
@handbury I remade the Census tract dataset and Prepped-Household-Panel.dta.
gisjoin is the Census tract.
HaveTract = 1 is an indicator for whether we have the census tract. It is 0 for panel year 2012-2016 always, and occasionally 0 before that because I make missing Census tracts if they are inconsistent with zip codes per PrepHouseholdCensusTracts.do.
So you can go ahead with the analysis we discussed this morning?

**Comment by handbury** [comment]
@huntallcott and @mollyschnell The results with tract x year (instead of zip x year) controls are the very similar for the disparity between the highest and lowest income group, but quite different (and not nearly as clean) for the disparities with other income groups.  The main plot is in a comment at the end of the lyx file, and attached.
[HealthIndex_Income_gisjoin_store_code_uc_x_yr_minexpsh50.pdf](https://github.com/huntallcott/FoodDeserts/files/2096566/HealthIndex_Income_gisjoin_store_code_uc_x_yr_minexpsh50.pdf)

**Comment by huntallcott** [comment]
@handbury thanks for this. It seems like it would be useful to determine what drives these differences. Just to clarify, did you use only the actual tracts for 2004-2011, or did you use the imputed tracts and zips for the full sample? 
It may also be useful to see how this looks with a nutrient score instead of the Health Index. This additional scores can be added in Code/DataPrep/Homescan/InsertHealthMeasures.do, and I can do this or if you get to this before me I would welcome it.

**Comment by handbury** [comment]
@huntallcott I used actual tracts for 2004-2011: the tract FE analysis is based on the "HaveTract==1" subsample.  Some of the difference is driven by this sample change (see the updated lyx file for the zip code fe plot using the HaveTract==1 sample), but not all. 

The HRS "nutrient scores" are calculated at the monthly frequency and only through 2011.  Since we're planning on adding something closer to the HRS nutrient score into the paper for robustness, it probably makes sense to include the something akin to the nutrient score in the new pipeline calculations, rather than merging the HRS indexes in.  The key changes are dropping sugar and F&V and implementing the squared loss function.  I'm travelling through the rest of the week but can look at this next week if you don't get there first.

**Comment by handbury** [comment]
@huntallcott @mollyschnell In the attached document, I replicated all of this analysis using:
1. the nutrient score in place of the health index and 
2. years of education (exp(lnEduc) - presumably the household head years of education since it's in round numbers) in place of household income - note that here I drop households with education < 10 years - there are only a few thousand of them and their scores look very odd).

The results for the nutrient score look similar to those for the health index except within stores, where perhaps we expect the nutrient score to be a bit weird since it's penalizing you for an unbalanced basket and we're looking at as little as 50% of a household's monthly expenditures.  The results look more normal again when you use years of education.

The weird patterns that we are seeing in the tract fixed effect analysis are still there with the nutrient score as the LHS variable but aren't as pronounced when looking at the education gradient.  I honestly can't think of why this would be the case at the moment.  It doesn't seem to have to do with the overlap sample issues since the overlap sample gradient is almost identical to the full sample raw gradient with both income and education (though it would likely be more different once we started aggregating income/education groups, e.g., to college/non-college as we do in HRS). 

Perhaps we can talk it over tomorrow morning?  Molly and I are talking at 9.  Are you around to chat in/around that time Hunt?

[WithinLocationAnalysis_AppendixFigures.pdf](https://github.com/huntallcott/FoodDeserts/files/2117008/WithinLocationAnalysis_AppendixFigures.pdf)

**Comment by huntallcott** [comment]
I can talk at 9AM. Call me at 617.256.9282? What if we use the meeting notes [here](https://docs.google.com/document/d/1QqaancZFShrx5ZY1-5nZne6v-LYs5E8sN4XXd_F4EmY/edit?usp=sharing)?

**Comment by handbury** [comment]
@huntallcott @mollyschnell I've implemented the changes that we discussed the other day (no longer normalizing relative to the lowest SE group; adding household size and age dummy controls to all income disparity analysis; calculating the >70K vs <25K disparity reduction).  The results are here and look similar to what we had before, though smoother.

[WithinLocationAnalysis_AppendixFigures.pdf](https://github.com/huntallcott/FoodDeserts/files/2127658/WithinLocationAnalysis_AppendixFigures.pdf)

The results are still odd looking within tract (Figure 3 in the appendix figures) so I replicated everything dropping the projection factor pweights and, for the within-store analysis, replacing them with calorie share aweights (closer to what we do in HRS).  This smooths things out quite a bit.  See the second attachment for these results.

[WithinLocationAnalysis_AppendixFigures_noprojweights.pdf](https://github.com/huntallcott/FoodDeserts/files/2127663/WithinLocationAnalysis_AppendixFigures_noprojweights.pdf)


@huntallcott FYI: to add the household size control, we need to keep and label the "household_size" variable in lines 189 and 203 of Code\DataPrep\Homescan\HomescanHouseholdDataPrep.do. Something happened when I tried to commit these changes since the code had been changed elsewhere while I was working on it.  I had to "recommit" a bunch of files excluding this file, which I believe discardedeh chanle I was out of the program.  Do you know where I can go to look at the recycle bin to figure out how to merge the changes across the two versions?

**Comment by huntallcott** [comment]
@handbury ,
Sorry for the delay. I was at a conference where email is not encouraged from Thursday night through yesterday. Here are some opinions:
1. Either of the figure 3's seems pretty OK to me.
2. I do lean toward using projection factors for these figures. 
3. I don't have a strong opinion on also weighting by calories. One thing to consider is 
`gen WeightedCalories = Calories*projection_factor`
and then weighting by `WeightedCalories`.
4. I've been using CalorieNeed as the measure of household size, as it is a nicer reflection of the amount of dollars the household has to spend on the groceries it needs. It probably doesn't make much quantitative difference whether we use that or household_size. Is it OK to use that or is there a strong tradition of using count of household members in this situation? If so then we should switch to household_size throughout the paper.
5. I am sorry about the committing problems. When things like that happen to me I usually curse and either use git stash or manually copy and paste files and then make sure my changes get to the master repo. I am not exactly sure about the recycle bin questions, and I'm sorry not to be of much help.

**Comment by handbury** [comment]
@huntallcott For the sake of consistency, let's stick with the projection factor weights and using calorie need as the measure of household size (though I will keep the label as "Household Size and Age Controls" to make it clear that we're using calorie need as a household size control).  I'll update the figures accordingly.

In response to your specific comments above:

1. Would you prefer to present the "within tract" results in the main text then?  My preference would be the "within zip" results, since they use the full sample of data.

2. The "without projection weights" analysis was useful to see that we can replicate the smooth graphical results presented in earlier versions of HRS but I see the argument for projection weights.

3. I would prefer to not weight by calories since there is probably a pretty strong mechanical connection with calories purchased and the health index.  In HRS we weighted by expenditure shares for the within store analysis to deal with households that aren't buying much at a given store.  This is also slightly problematic (healthy foods tend to cost more).  I prefer the new approach of limiting the within-store sample to household-store-year observations that reflect over half of the household's calories in that year.

4. There isn't a strong tradition of using household member count but instead an equivalence scale adjusted income measure.  We don't want to do the latter because it changes the x-axis, so we have to do something non-standard here regardless. 

5. I'll look into git stash for future committing problems.  Thanks for the tip!

**Comment by huntallcott** [comment]
@handbury , responses in your numbering system:
1. What about using CTractGroup, which uses the full sample but is mostly tracts? I know I've gone back and forth about this, but I also note that these are the same location IDs that we are using in the store entry analysis (household x CTractGroup fixed effects) and in @mollyschnell's #11, so it seems internally consistent and easy to explain if we stick with those.
2. Great, agree re: using projection_factor.
3. Great. Agree to not weight by calories and limit sample as you say.
4. OK great.
Thanks again.
Once you're happy with this you can write it into the lyx file at your leisure.

**Comment by handbury** [comment]
@huntallcott I think that its much cleaner to use zip codes or tracts, rather than a mix of the two for this analysis.  There will be households who live in the same zip but we observe the tract for one and not the other, for example, so we will effectively lose them from the identifying sample here.  It makes more sense in the entry and time series analysis where all we are pulling from the zip/tract is the location of the household.

**Comment by huntallcott** [comment]
@handbury I don't feel strongly and I defer to you. But a quick follow-up: if the main concern is about the size of the identifying sample, isn't CTractGroup strictly better than tract? And if we are worried about the size of the local market area, isn't tract materially better than zip?
Again, I don't feel strongly and I defer to your expertise, but I just wanted to understand fully.

**Comment by handbury** [comment]
@huntallcott @mollyschnell, I have inserted this within-location analysis after the event study results - making that section a broader "reduced-form" section with the event studies and the within-location results.  The idea here being that the event study analysis gives us an estimate of short-run impacts while the within-zip/store analysis tells us about longer run effects.

Another option that might make for a smoother narrative (and less changes to the intro), would be to put these results into the stylized facts, simply making the point that the disparities documented there exist even within locations and within stores.  To write things this way, I think that we would need to re-order the stylized facts, presenting the availability disparities before the consumption disparities and then finally using these within-location "stylized facts" as the basis questioning whether we believe that there is really a strong role for the availability in driving consumption in this circumstance.

@mollyschnell could you think about what works best when you are reviewing in the coming week or so?

**Comment by huntallcott** [comment]
@handbury this is brilliant. Some thoughts:
1. I don't feel strongly but I have the intuition that this is more than a "stylized fact." This is an insightful empirical strategy generating a pretty tight bound, and we sell ourselves short if we put it as a "stylized fact." You can't do this with NHANES.
2. I moved the figure to the end with the other figures and made a few minor edits but generally I think it's perfect.
3. If we are moving to using the HEI you should re-run this analysis. I'll remake the data overnight. You should look for the variable lHEI_per1000Cals. In the code, I would call the variable $HealthVar (defined in SetGlobals.do), so we can easily change back if, for example, HEI works out poorly in the demand estimates.
4. Can you look at the table note one more time? I think that you're taking residuals and using household average income now, as in Figure 2, but the table note reflects an old approach? You can cut and paste from the table note in Figure 2.
5. One other minor expositional thing to take a look at: you'll see in sections 4.1 and 4.2 I report an actual number for the bound, e.g. 3% instead of "three quarters". I might have a slight preference for us reporting that exactly so that it's clear and consistent.
* If you're interested, you could write the code to automatically output that bound from stata and pull it into lyx. See, for example, in Code/Analysis/StylizedFacts/StylizedFacts.do line 57, there is a global created for the quartile difference in healthy eating. You can then use that global to automatically output a share of difference explained, as I do in Code/Analysis/ReducedForm/EntryRegs.do lines 376-420. Then you can see that the lyx file in Section 4.1 automatically pulls this in using an input command.

This will take a moment to code but reduces marginal cost to zero if we change some data prep routine or whatever. Up to you but I've found this really helpful in other papers.

Thanks again.

**Comment by huntallcott** [comment]
@handbury I re-made the HHxYearxStore data and inserted the linear HEI and also the standard (non-linearized) HEI, which is called HEI_per1000Cals.

**Comment by handbury** [comment]
Great.  Thanks Hunt.  I’ll process this morning.

On Aug 13, 2018, at 7:49 AM, Hunt Allcott <notifications@github.com<mailto:notifications@github.com>> wrote:


@handbury<https://github.com/handbury> I re-made the HHxYearxStore data and inserted the linear HEI and also the standard (non-linearized) HEI, which is called HEI_per1000Cals.

**Comment by handbury** [comment]
@huntallcott:

I'm having trouble compiling in the older version of lyx but have made these updates so figured worth committing.  Hopefully the section will compile (and look fine) at your end while I figure out the issue.

On the footnote: to keep the income axis easy to interpret, the income brackets were kept constant (i.e., not residualized) and we control for household size and age (with CalorieNeed and i.AgeInt) in the regression where the mean index by income bucket is calculated.  It looks like you do the same in the Stylized Facts figure, except using HHAvIncome (constant average household income) and HouseholdSize, i.AgeInt, and i.panel_year as controls.  I have adjusted to match this approach for consistency.

Thanks for the suggestion on the automatic stat import.

We have the figure and stats for lHEI, healthindex and nutrient score and we can switch between then by just changing the figure/stat files that we read in.

Thanks,

Jessie

**Comment by handbury** [comment]
@huntallcott: I had trouble committing the updates to the within-location section/figure in the draft (when I pulled the recent draft file  I got a "file corrupted" error when trying to open it).  I think that this is because someone is working simultaneously.  I'll try to put these changes in again later but in case you want to get things together before then, I have committed just the section and figure that I updated in the file FoodDeserts_withinlocupdates.lyx file.

Sorry for the inconvenience!

**Comment by huntallcott** [comment]
@handbury great and thanks for working on this. Quick notes:
1. I can compile the main lyx file. I had to change some hard references on your computer to relative references, and the other thing to note it that lyx won't compile if we have a latex commend box inside a footnote. So I fixed those things and am able to compile.
2. I did not merge in the other files currently in Code/Lyx.

**Comment by huntallcott** [comment]
@handbury @mollyschnell 
I looked again at the lyx files. @handbury your commit from this morning accidentally overwrote the changes that Molly and I had made over the past few days. I am going to revert the lyx file to where it was as of last night. I then inserted FoodDeserts_withinlocupdates.lyx into section 4.3 and figure 8.
So I think we are good. 
Maybe hold off on more edits for a few hours? I need to make a few changes before I head off for the wedding.

**Comment by huntallcott** [comment]
@handbury I continue to love what you wrote here.
I just have two minor things:
1. can you figure out how to get the figure to compile in order with the rest of the figures? I tried and failed. You can see that right now it's compiling at the end of the figures.
2. it might be worth changing either the dots or the diamonds to crosses so that these figures will be readable in black and white.

**Comment by handbury** [comment]
@huntallcott 

Welcome back, congratulations and sorry for not getting to these while you were away!

1. I think that this is fixed (I reset the figure placement option).  I can't compile the draft currently because it can't find some numbers (I presume that it is in process) so I haven't made this change. To make the within location figure compile in order with the other figures, I think that we just need to make it have the same placement option selected.  In case there's an error in what I did, I did the following: right click the "float: Figure" button and select "Settings..." then select "Use default placement" (the option selected for the other figures).  

2. I've changed the within-location option to an X and kept the raw mean as a filled blue circle (to match the earlier plot).

**Comment by moosedube** [comment]
@handbury @Rahkovsky @mollyschnell: Spent the day reading the paper from start to finish and making edits. Section 4.3 is a nice complement to the two event studies. But, I admit that I am surprised/confused why controlling for zip has a much bigger effect on the gap than controlling for store. One thought is that the store assigned to a HMS trip is measured with error since most HMS panelists only report the chain. HMS then assumes a household shopped in the closest store to their home address from the chain. This could attenuate the effect of “store”.  Have we tried controlling for chain instead of store (since chain should be measured more accurately). 

If there is attenuation bias, then I guess we turn to the zip results. But now the LR effects seem quite big to me: within-zip-code analysis reduces the nutrition gap by almost 30%.  We need to reconsider our wording in the intro in this case, where we insinuate that they are not all that big.

I think the selection story you guys suggest makes sense as a caveat to calling this the actual "LR effect of availability."  I wonder if we can see some evidence of this. In Figure 6, it looks like the gap is closing largely due to the top-income households: rich people eat worse food once we look within-zip. I wonder if this is preference (i.e., your selection story) or supply.  Do these neighborhoods tend to have stores with fewer healthy options? For instance, are these typically very urban neighborhoods with smaller supermarkets offering less variety (i.e., potentially less health options)?  Within a given chain, a supermarket in Lincoln Park (wide income range) is smaller than one in a north suburb (mostly upper-income). So the zip-level analysis would tend to use the former stores than the latter.

**Comment by Rahkovsky** [comment]
@moosedube I think your story about measurement error is plausible, but I am not sure if it can generate large effects. Stores of the same chain tend to carry similar products. The difference between Safeway close to work and Safeway close to home, should not be too large. 

Figure 2 shows that higher income areas have monotonically healthier food. I think the selection story is more likely. Another interpretation of the zip-code results is that low-income consumers in the rich zip-codes have very healthy diets, possibly eating better than the high-income consumers living there. I think this is where the selection happens.

**Comment by handbury** [comment]
@moosedube, I've looked into this further.  The results using chain fixed effects are very similar.

On the second point, the set of “overlap” zip codes typically have larger stores with lower health indexes (either ADD or HEI), however, this shouldn't affect the results since, in **both** the baseline and within-zip estimates, we limit our attention to the "overlap" sample (the set of households that reside in zips with sample households from other income groups).  This controls for the fact that rich households living in zip codes with poor households might have different preferences to the overall population as well as for the fact that these households might have access to different sets of stores.   

The restriction that households of "other income groups" reside in the same zip (or, in the within-store analysis, are shopping in the same store) is quite weak when we are looking at income bins (i.e., in the plots).  Many zips make it into the sample because they have panelists a single income bin away from one another shopping in them.  When we run the quartile bin analysis, we require that panelists from more than one quartile reside in the same zip (or shop in the same store) and we get that the gap between the high- and low-quartile households in these zips closes by around 25% when you compare within zip.  Note that if we are more restrictive about the overlap - limiting the sample to panelists in the top and bottom quartiles and then only looking at panelists living in zips with one of each of these groups residing in them the gap closes instead by only 13%.  With this sample limitation we are really looking at a quite odd slice of the sample (perhaps something that you might see around Hyde Park).  In HRS, our overlap sample was tracts with both college and non-college panelists - we could do something similar here by looking at the disparity between above and below median income panelists, but that would move away from what is elsewhere in the draft.

**Comment by huntallcott** [comment]
Hi @handbury @moosedube @mollyschnell ,
Here is my two cents. 
A. I like this analysis, even though the zip code version seems to be generating a looser bound. I think @moosedube is worried about the looseness of this bound, but I still think that 30% is informative and interesting, even if it's our loosest bound, and in any event, the store version is much tighter.
B. I think we want to at least give a hypothesis as to why the zip fixed effects seem to attenuate the nutrition-income slope more than the store fixed effects do. Right now we just say that we are surprised. 
C. The biggest-picture thing is that I think we should follow the exact same formula mechanically in 4.1, 4.2, and 4.3. I think that formula is:
1. Give the motivation/overview.
2. Discuss data and the estimating equations.
3. Clearly explain the selection bias concerns and clarify that selection bias operates in a way that makes our argument stronger. (in 4.1, this is brief, and I think it's fair to say that they are minimal.)
4. Present the figures.
5. Present the table. (Not relevant for 4.3).
6. Give the quantitative bound on contribution to the nutrition-income relationship.

You guys are all better writers than me, but I point this out because (i) I see these subsections covering these issues in slightly different order, and (ii) I see some subsections covering these issues twice, e.g. in 4.3 we discuss selection bias in several different places. 

@mollyschnell, given your skills and the great work you did so far, I was wondering if you would be willing to look through all of Section 4 once more and make sure this order is followed? Also, at this point I think it's safe to accept all the changes and end with a clean copy with no notes or changes. 

Then after that I can do a read and also will add these robustness checks I'm finishing, which are mostly just footnotes for the referees.

Is that OK? If so, @mollyschnell let me know your anticipated timing so I can coordinate when I'll put in the additional tables/robustness checks.

**Comment by mollyschnell** [comment]
@huntallcott: Happy to take another look! I won't have the chance until later today, so feel free to be in the draft this afternoon. Why don't you let me know when you're out and then I'll hop in?

Molly

**Comment by handbury** [comment]
@huntallcott @mollyschnell, I don't know if you have done any more work writing this section up, but in case not, I thought it worth doing some further reconciliation to help inform the "puzzling" result with the within-store reduction being so small vs. the within-zip reduction.  

First, when we run the within-store analysis, we limit our attention to households that purchase more than 50% of their calories in the same store.  When we look at the disparity within-zip using this sub-sample, we see the same puzzling large bound from the within zip analysis vs within-store (note that the numbers are slightly different since I changed the identification sample from being stores with shoppers with two income quartiles, where it was previously stores with shoppers from two income bins):

*Q4 vs Q1 gap from = .61 to .48 st dev with zip_code fixed effects
*Q4 vs Q1 gap from = .64 to .62 st dev with store_code_uc fixed effects

I ran the same checks with retailer code fixed effects and found that the sample changes, since we are pulling the retailer code from the RMS/Stores-Prepped dataset which presumably only contains the (typically larger) chains in the RMS data.  In the sub-sample of households who spend more than 50% of their calories in these larger chains, the Q4-Q1 gap in the health index is smaller but the reduction in the gap with zip and store FEs look similar:

*Q4 vs Q1 gap from = .48 to .35 st dev with zip_code fixed effects
*Q4 vs Q1 gap from = .48 to .35 st dev with store_code_uc fixed effects
*Q4 vs Q1 gap from = .5 to .44 st dev with retailer_code fixed effects

The reduction in this sub-sample is approx 26% for both zip and store FEs, less for retailer FEs..

As I mentioned above, the reduction in the gap is lower when we limit our attention to households in Q1 and Q4. That is, identify directly with households that live or shop in settings with panelists from both ends of the income distribution, comparing only purchases of Q1 and Q4 panelists who live/shop in the same place as one another (as opposed to also identifying indirectly using the differences between purchases of households in the intermediate quartile pairs - Q1-Q2, Q1-Q3, Q2-Q3, Q2-Q4, and Q2-Q3 - who live/shop in the same place):

- in the sample with both big and small retailers:
*Q4 vs Q1 gap from = .53 to .48 st dev with zip_code fixed effects
*Q4 vs Q1 gap from = .68 to .66 st dev with store_code_uc fixed effects

- sample where we observe the retailer code:
*Q4 vs Q1 gap from = .41 to .34 st dev with zip_code fixed effects
*Q4 vs Q1 gap from = .4 to .34 st dev with store_code_uc fixed effects
*Q4 vs Q1 gap from = .51 to .45 st dev with retailer_code fixed effects

I'd be curious to hear what you guys think of whether the indirect inference (using both the Q4-Q1 difference as well as the other inter-quartile differences to identify the Q4-Q1 gap) makes sense.  I think that it's reasonable to identify using both indirect as well as direct comparisons when we're using the full sample, but perhaps less reasonable to do so in our "within-zip" and "within-store" analysis, since we could in theory be identifying the Q4 vs Q1 difference in purchases in the "same retail environment" without actually observing any purchases of Q1 and Q4 households shopping in the same store or living in the same zip.

Let me know if you want to hop on a call to discuss.

**Comment by mollyschnell** [comment]
@handbury: Thanks for looking into this! 

I think it would make sense to require that we observe households in both Q1 and Q4 in a given environment (rather than just two households in different income quartiles, so that we're not extrapolating to the Q4-Q1 difference). Under that sample restriction, do the estimates vary greatly depending on whether we include Q2 and Q3 households in the estimation?

It's interesting that limiting the sample to RMS stores causes the store fixed effects to reduce the gap by so much more. Just to check, we weren't limiting the sample to RMS stores in HRS, were we? The RMS results seem easier to explain (since we at least get similar bounds between the two analyses), although I don't have a good intuition for why households sorting into larger retailers would have more correlated preferences. Maybe larger retailers attract customers from a broader geographic range, so that it's a more selected sample than the mix of people swinging by a neighborhood store?

**Comment by moosedube** [comment]
Indeed, thank you for doing this analysis.  Quick question: when you condition on retailer code, why do you only focus on RMS stores?  We should observe the retailer code in the HMS sample too.  It seems like there are two issues here. First, we want to conduct the within-place analysis using people who shop in the same chain, as opposed to the same exact store.  This is important since store ids are potentially measured with error (chain ids are not).  Second, we want to conduct the analysis with people who shop only in large-format stores (using RMS as a proxy). Incidentally, we can probably get better measures of retailer size than RMS participation. For instance, we can use the outlet codes to focus only on large supermarkets (unless we are already doing that).  We can also focus on the subset of supermarkets that appear in at least XXX scantracks, where XXX is selected to capture chains with large geographic coverage (which I suspect is what you were getting at with the RMS data).

JP

**Comment by handbury** [comment]
@moosedube:  I merged the retailer codes in from the Externals/RMS/Stores-Prepped.dta dataset that I had assumed only contained RMS stores.  It contains all of the stores in the $Nielsen/RMS/`year/Annual_Files/stores_`year'.tsv datasets.  Its possible that these raw datasets also include the stores in the HMS data that aren't in the RMS data but this seems unlikely. 

@mollyschnell: We were using the retailer_code from the HMS data in HRS, so didn't have any RMS sample limit there.

Also, when we limit the stores/retailers/zips that we look at to those with both Q1 and Q4 households, but then include Q2 and Q3 households from those locations in our sample the results are quite similar to when we only look at Q1 and Q4 stores.  Note that this isn't mechanically the case: in the "RMS" store case, the vast majority of RMS retailers have panelist households from all 4 quartiles shopping in them and in the "all store" case, over 80% of the stores or zips that have representation from both Q1 and Q4 also have representation from at least one other income quartile.  I think that it indicates that the fact that we see the gap dropping when we look at stores/zips with both Q1 and Q4 represented gives us a selected sample of panelists who have more "average" tastes in general (i.e., people who haven't selected into really segregated zips/stores).  Perhaps this type of selection also explains why the panelists shopping in RMS chains also exhibit more correlated purchases (both across and within chains). 

I don't see any reason to limit our attention to large, national retail chains but, for the purpose of understanding why the results look different when we limit our attention to RMS store shoppers, it will be useful to look at the within-retailer code results for all retailers (not just the RMS ones).  Then we can run a more general "large geographic coverage" sub-sample test, like the one you are suggesting @moosedube.

So, @huntallcott , would you mind re-running the code that creates the HHxYearxStore.dta dataset so that it includes the retailer_code variable provided in the raw HMS trips data?

Thanks!

**Comment by moosedube** [comment]
The raw trip files for HMS contain store and retailer codes. So if these were not retained at the build stage for this project, we can easily re-run this to make sure you have all the retailer codes for all HMS trips (not just those to RMS stores).

JP

**Comment by huntallcott** [comment]
@handbury @mollyschnell @moosedube ,
The file $Externals/Calculations/Homescan/HHxQuarterxRetailer.dta is already collapsed by retailer, so you can look at that if you want. I am also re-making $Externals/Calculations/Homescan/HHxYearXStore.dta to collapse by retailer x store and retain retailer_code. That should take a few hours.
This is all in Code/DataPrep/Nielsen/TransactionDataPrep.do, so you can modify in the future if this is unsatisfactory.
Thanks again for looking into this.
-Hunt

**Comment by handbury** [comment]
Thanks @huntallcott. I'll wait for the HHxYearxStore dataset to complete and get back to everyone on this early tomorrow.

**Comment by handbury** [comment]
The results look as expected when using all retailer codes.
The top-bottom quartile gap looks similar whether looking at the HH-year or HH-store-year data (0.56, 0.59 standard deviations, respectively). The gap closes  to 0.43 when controlling for zip FEs and 0.41 when controlling for retail chain.  This is a drop of about 25% within zip and 30% within retail chain.  This is  estimating using households from all 4 income quartiles.

When we limit to households from the top and bottom income quartiles, the results are similar within-chain (the gap closes from 0.59 to 0.42) but quite different for within-zip, where we lose a half of the households in our sample because there is much more income sorting across residential zips than across retail chains.  In the smaller "identifying sample" of households living in zips with both high and low income residents, the raw high-low gap drops from 0.57 to 0.50 and then only to 0.44 when we look within zip.

@mollyschnell: I have edited the relevant section of the draft to reflect the first set of results above (with households from all quartiles used to identify the gap between the top and bottom quartiles).  Let me know if you want to switch.  Its a tighter bound (for the within-zip results) but mostly because of the sample selection which might explain more discussion than its worth (given that the within-retail chain results don't change).

**Comment by handbury** [comment]
I’m re-running the code dropping those retailers now.

I will commit and then play around with looking within chain*zip or chain*county.  I think that the chain*zip will run us into small overlap sample issues but the latter one should work.

**Comment by huntallcott** [comment]
chain-state or chain-dma are also fine, although county I guess is nicer both because it's smaller and because we use county a lot elsewhere.
thank you!

**Comment by handbury** [comment]
The results are pretty similar whether you include chain or chain*household location interactions, up to when you use chain*HH zip controls

Controlling for chain*year FEs reduces the IQR of the health index from 59 to 41 std deviations and we lose less than 1% of observations by restricting ourselves to chain*years where we observe HHs in both the top and bottom quartiles.  With chain*HH state*year FEs, the IQR drops from 59 to 39 std deviations, with about 2% of observations dropped by restricting ourselves to chain*HH state*years where we observe HHs in both the top and bottom quartiles.  The results are very similar for chain*HH county*year FEs.  

With chain*HH zip*year FEs, within the "identifying" sample (chain*zip*year's where we observe both a top and bottom quartile household), the IQR drops from 49 to 34 standard deviations.  This is the same percentage drop, all yield a bound of 30%, just a different baseline IQR.  The drop in the baseline IQR is due to the identifying sample being much smaller.  We drop over 50% of observations from chain*zip*year's without a household in both the top and bottom income quartiles.  

Since the results are so close to our current results when we look within chain*county*year, we could either just state that this is the case in a note or switch the reported results to be looking within store*county.  

Note that controlling for chain*HH location is slightly different to controlling for chain*store location.  It probably serves a similar purpose at the county level, but below that (e.g., at the zip level) the chain*HH location is controlling for what people can buy in the chain (presumably the same outlet) interacted with the rest of their tract-specific retail environment while the chain*store location is controlling only for what people can buy in the outlet.  

I don't think that we should add any results here, beyond a sentence saying that the same results are observed for , but would you guys be in favor of subbing out the chain*year for chain*HH county*year or chain*HH zip*year results?  If so, let me know and I'll run things to produce these plots.

Thanks!

**Comment by huntallcott** [comment]
@handbury many thanks for doing this.
I think I'm in favor of chain by county instead of chain only, but I am open to others' views!

**Comment by handbury** [comment]
@huntallcott Sounds good.  I committed the results with chain x HH county so that we can put them into the draft we want them.

**Comment by handbury** [comment]
@huntallcott:  @mollyschnell and I looked into the questions you had in section 4.3.  

It made sense that Figure 1 didn't match the "Raw" points on the top left of Figure 6 because:
(i) the "Raw" points are based on the "identification" sample. 
There were other reasons that the points didn't match, however:
(ii) Bin scatter was not estimating the residualized means with the same spec that we had written down in section 4.3.  Instead, binscatter residualizes lHEI and HHAvIncome from the controls (call these variables lHEIresid and HHAvIncomeresid) and then calculates the average of lHEIresid in 15-ile bins of HHAvIncomeresid. Do you have any reason to prefer this way of calculating the residualized means above the specification presented in Figure 6?  We could implement the same methodology for Figure 6, but it would require more explaining than the regression equation.
(iii) Figure 6 uses pweights instead of aweights.  I think that this is the correct weight for sampling weights (see: https://www.cpc.unc.edu/research/tools/data_analysis/statatutorial/sample_surveys/weight_syntax).  The difference in the estimates based on this alone, though, is small.  This should be an easy switch in either Figure 1 or 6.

I think that it will make it easier to digest if Figure 6 shows the same data as in Figure 1 (presuming we reconcile ii and iii) and then switches to the identification sample before residualizing from market fixed effects.  To that end, I think it worth reconciling the two figures and presenting three sets of points on Figure 6: the "Raw" full sample means, the "Raw" identification sample means, and the "Residualized" identification sample means.  I've made plots that do both so we have flexibility here.  

On the >50% calorie share sample control in the "within chain" analysis, that was to avoid the criticism that the poor households that we're observing in whole foods or rich households in the dollar store are only "stopping in" for the odd item.  Limiting ourselves to household-chain bundles that reflect the majority of a households' purchases in that year both dealt with this concern and conveniently brought the unit of observation back to the household-year level.  In any case, I'm producing results without this restriction for the purpose of comparison - more extra results that we can delete when we pick a final spec.

I'll commit these extra results once they're out later this evening.

**Comment by huntallcott** [comment]
@handbury @mollyschnell thanks, and thanks for entertaining my questions on that section. Responses:
(ii): I like the binscatter for Fig 1, and I also like the idea that we do the same thing in Fig 6 AND Fig 1. In keeping with that, another way to do this in a simple way is just to write the linear regression
Y = \alpha w_i + \gamma X_it + \mu + ...
And then say we're showing a binscatter of Y and w in that regression. I think this would be fine to explain.
Aside: please do use \alpha w_i if you write it that way, as that matches what we do in the Gelbach decomp later.
Aside number 2: what if you just use the binscatter command if you're not already?
iii. Binscatter doesn't allow pweights, and it gives the exact same regression line with fw and aw. I think that for least squares and binscatter, the weights only affect the standard errors. 

Excited to see the new figures.

Thanks again.

**Comment by handbury** [comment]
Thanks @huntallcott.  We can implement the binscatter procedure for the market fixed effect approach (though I’m not sure it would be efficient to use the binscatter with the market fixed effects unless we were sure that binscatter used something efficient like areg underneath).

I see what you mean in describing the binscatter of y and x as being those variables “from a regression.”I I guess binscatter does something like what areg does - residualizing y and x from controls and then regressing (or plotting, in the case of binscatter) yresid against xresid.  But will this also be transparent to a reader?

One issue that is raised when thinking of implementing binscatter without and then with market fixed effects is that the bins and their “mean income” be change between specs.  In the first spec x will only be residualized from controls while in the second, x will be residualized from market fixed effects as well as controls.  Accounting for that will make for a messy plot.  There are two ways around this:

1. Find yresid_market (residualized from controls and market fixed effects) and then to plot for bins of the original xresid, but that is no longer  “plotting y and x from the (market fixed effect) regression”.

2. Do a non-parametric fit (lploy) of the y residuals against the w residuals.   This would be my preference if we’re doing the binscatter approach.

Part of the reason that I like using income bin coefficients from Y = \delta_k I(w_i in bin k) + \gamma X_it, as we have been doing in Figure 6, is that it then seems reasonable to plot these coefficients against the mean of raw w_i in each bin (and creating bins based on raw w_i).  Any chance you’d be willing to switch to this approach for Figure 1?

If not, then I think that we should switch to non-parametric representation to avoid having to explain why the bin income means move around.

On Oct 23, 2018, at 8:02 PM, Hunt Allcott <notifications@github.com<mailto:notifications@github.com>> wrote:


@handbury<https://github.com/handbury> @mollyschnell<https://github.com/mollyschnell> thanks, and thanks for entertaining my questions on that section. Responses:
(ii): I like the binscatter for Fig 1, and I also like the idea that we do the same thing in Fig 6 AND Fig 1. In keeping with that, another way to do this in a simple way is just to write the linear regression
Y = \alpha w_i + \gamma X_it + \mu + ...
And then say we're showing a binscatter of Y and w in that regression. I think this would be fine to explain.
Aside: please do use \alpha w_i if you write it that way, as that matches what we do in the Gelbach decomp later.
Aside number 2: what if you just use the binscatter command if you're not already?
iii. Binscatter doesn't allow pweights, and it gives the exact same regression line with fw and aw. I think that for least squares and binscatter, the weights only affect the standard errors.

Excited to see the new figures.

Thanks again.

**Comment by huntallcott** [comment]
@handbury , thanks.
Are you sure that in practice you get a messy plot when you use different fixed effects approaches in binscatter?
One other option to consider, and @handbury maybe this is equivalent to what you're saying, but what if we create a new variable called HHSampleAvIncomeBin in the household-panel data, that has the mean income within each of the 15-tiles of HHSampleAvIncome? Then we do binscatter, discrete using that. I'm not sure if that helps but maybe also worth considering.
I guess for me a next step would be to just see the plots you're generating.

**Comment by handbury** [comment]
@huntallcott - I've rewritten the code to process using binscatter.  The changes in the residualized income bins are small between the full sample and the identification sample.  The changes are - unsurprisingly - much larger when we additionally residualize from market fixed effects (the 1Q/4Q mean income going from $18K/$114K in the data residualized from demographic controls only to $28K/$94K also residualized from chain x county x year fes).  This is happening because the chain mean income is being removed (households sort into chains by income). I don't think that this provides the right x-axis.

So I agree that we should fix the income bins (as we were doing in our original analysis).  I'll just pull the income bins residualized from controls from the base sample and then use these throughout.

Results TBD shortly.

**Comment by handbury** [comment]
The results using the binscatter method are very different.

I think that this is because x is not actually controlled for in the regression that residualizes y and x from the controls and absorb variable.

In our case, it runs Y = alpha + gamma X_it  (+ delta_m) + epsilon and w = alpha' + gamma' X_it (+ delta'_m) + epsilon' and the does a binscatter of (alpha + epsilon) against (alpha' + epsilon').

I think that, instead, we want it to run the y regression including w.  The slope of the points is the same, when plotted against the relevant w BUT (as I mentioned above in the x-case) the range of both the x and y variables (after being residualized from controls + market fixed effects) is lower.

The results from binscatter and our old method (showing the ID sample vs full sample comparison) are both below:
Bin scatter:
[bscat__Income_zip_code_chain_x_county_x_yr_minexpsh50ID.pdf](https://github.com/huntallcott/FoodDeserts/files/2511249/bscat__Income_zip_code_chain_x_county_x_yr_minexpsh50ID.pdf)
Old method:
[lHEI_Income_zip_code_chain_x_county_x_yr_minexpsh50ID.pdf](https://github.com/huntallcott/FoodDeserts/files/2511256/lHEI_Income_zip_code_chain_x_county_x_yr_minexpsh50ID.pdf)

**Comment by handbury** [comment]
Also, @huntallcott, the "discrete" workarounds we thought of to align bins across the specs don't work with controls (that is, neither controls() or absorb() work correctly with the xq() or discrete options).  As a workaround, I tried residualizing the y variable independently from the control and then run bin scatter without controls and absorb but this isn't a correct application of the method (which relies on both the y and x variables being residualized - and residualizing the "discrete" x variables from the controls and absorb variables leaves you with the same distribution of the quantiles as you get when you allow binscatter to create quantiles from the raw w data residualized from controls and market fixed effects).  

We would hit the same issue if we created HHSampleAvIncomeBin based on raw income and then ran the figure 1 binscatter on that.  

So it seems that our choices are between using binscatter having to explain the different x-axis scales and our original approach.

I see how convenient bin scatter is for other applications (e.g., Figure 1) but it is clearly complicating things for Figure 6.  @huntallcott: do you have a substantive reason to prefer binscatter? If not, I think that most sensible way for us to be using the same approach in Figures 1 and 6 is to switch to plotting coefficients from "reg `outcome' ibn.HHSampleAvIncomeBin $SESCtrls i.panel_year [aw=projection_weight], nocons" in Figure 1 - rescaled to make sure that the plotted coefficients take the same mean as the outcome variable.

**Comment by huntallcott** [comment]
@handbury thanks. Who knew that binscatters were so complicated. Thanks for digging in to this.
I'm in favor of your preferred approach for Figure 6. Great.
I'm also in favor of the binscatter for Figure 1, and in any event we use binscatters elsewhere in the paper. So my proposal is just to have that inconsistency between how Figs 1 and 6 are produced. I think that this is totally tolerable.
Full steam ahead -- sound like the next step is to clean up the mess I left you in 4.3 if you're willing to?
Thanks again.

**Comment by handbury** [comment]
@huntallcott - I worked on 4.3, taking out the language comparing the baseline here with Figure 6 and putting the sample definitions up front. 


@mollyschnell - let me know if you can think of better organization/wording to streamline this section back to its original 1+ page length. 

The retail chain x county market spec is a bit awkward to describe in parallel with the zip code market spec.

**Comment by mollyschnell** [comment]
@handbury: Thanks for this! 

I agree that it's a bit awkward to discuss the two specifications in parallel. It would be easier if "m" were one dimensional (i.e., a zip code or a store), but there's not a great way to say what Y_imt is  when m = store x location (note that we currently just don't define m in the text). 

@huntallcott, any ideas?

**Comment by huntallcott** [comment]
Hi @mollyschnell @handbury,
I am actually not against the way it's done in the version I just pulled. It also parallels 4.2, where we introduce m and say that m could be either a county or zip code, then have one equation covering both "market" definitions.
I would just say that a market is is either a zip or a chain-location pair, and proceed as you do.
If I'm missing something, that's fine and proceed as you see fit, but your writing looks great to me right now.

**Comment by huntallcott** [comment]
@mollyschnell @handbury thanks for your great work here. Closing this issue.
